### ISSUE TITLE
Persisted jobs are not removed anymore in v5.0.1

### ISSUE THREAD

**Issue by Narayane** [lucas34/SwiftQueue#367]
Hi @lucas34,

I have just tested v5.0.1 and I'm sorry but relative bug is still present... Persisted jobs are not removed from "persistence" if app has been killed before. There are still restored again and again... 😕

**App launch logs with SwiftQueue v5.0.1**: 1 print of _insert in queue_, 1 print of _restore in queue_ for persisted jobs and only one print of _remove from queue_ for StopRideJob which is the only one not persisted. Persisted jobs are restored, still not inserted and not removed

16:03:59.344 💜 VERBOSE LRJobPersister.init():17 - create a LR job persister v5
16:03:59.344 💜 VERBOSE LRJobPersister.restore():26 - restore
16:03:59.345 💜 VERBOSE LRJobPersister.restore():34 - restore in queue
16:03:59.347 💚 DEBUG AutoRenameRideJob.init():64 - create a job to auto rename ride 3
16:03:59.350 💚 DEBUG StartRideJob.init():103 - create a job to start ride 3
16:03:59.350 💚 DEBUG PauseRideJob.init():65 - create a job to pause ride 3
16:03:59.351 💚 DEBUG ResumeRideJob.init():66 - create a job to resume ride 3
16:03:59.352 💚 DEBUG StartRideJob.onRun():110 - run the job to start ride 3
**16:04:00.204 💚 DEBUG StopRideJob.init():88 - create a job to stop ride 3
16:04:00.204 💜 VERBOSE LRJobPersister.put():42 - insert in queue**
16:04:00.562 💚 DEBUG StartRideJob.onRemove():144 - remove the job to start ride 3
16:04:00.563 💙 INFO StartRideJob.onRemove():150 - ride 3 has been started
16:04:00.563 💚 DEBUG PauseRideJob.onRun():72 - run a job to pause ride 3
16:04:01.197 💚 DEBUG PauseRideJob.onRemove():108 - remove a job to pause ride 3
16:04:01.198 💙 INFO PauseRideJob.onRemove():114 - ride 3 is paused
16:04:01.198 💚 DEBUG ResumeRideJob.onRun():73 - run a job to resume ride 3
16:04:01.715 💚 DEBUG ResumeRideJob.onRemove():109 - remove a job to resume ride 3
16:04:01.715 💙 INFO ResumeRideJob.onRemove():115 - ride 3 is resumed
16:04:01.716 💚 DEBUG AutoRenameRideJob.onRun():72 - run a job to auto rename ride 3
16:04:02.104 💚 DEBUG AutoRenameRideJob.onRemove():119 - remove the job to auto rename ride 3
16:04:02.105 💙 INFO AutoRenameRideJob.onRemove():125 - ride 3 has been auto renamed
**16:04:02.105 💚 DEBUG StopRideJob.onRun():96 - run the job to stop ride 3
16:04:02.248 💜 VERBOSE LRJobPersister.remove():54 - remove from queue
16:04:02.249 💚 DEBUG StopRideJob.onRemove():130 - remove the job to stop ride 3
16:04:02.250 💙 INFO StopRideJob.onRemove():136 - ride 3 has been stopped**

_Originally posted by @Narayane in https://github.com/lucas34/SwiftQueue/issues/362#issuecomment-739980214_

**Comment by arthurdapaz** [comment]
This is true. Persisted Jobs are not being removed. I had to manually adjust it on my own fork :(

**Comment by lucas34** [comment]
@arthurdapaz  Can you share your fork or open an MR ? How did you solve the issue ?

**Comment by arthurdapaz** [comment]
@lucas34 I'm actually investigating the issue right now. I didn't push it yet. It seems like it happens on very specific situation. For example: If my job is constrained to internet access... (so it is persisted)... I close the app (and then we have internet again, the job will be looping every time the manager is initialized)

**Comment by arthurdapaz** [comment]
@lucas34 for some reason, the persisterConstraint can't be retrieved (it's **NIL**). As you can see on my console screenshot:

<img width="872" alt="Captura de Tela 2020-12-19 às 23 48 48" src="https://user-images.githubusercontent.com/2267843/102703979-ca151f00-4254-11eb-8220-0c480d3c200a.png">


I couldn't debug those generic functions to see what they are making under the hood:
```
public func getConstraint<T: JobConstraint>(_ operation: JobInfo?) -> T? { ... }
public func getConstraint<T: JobConstraint>(_ constraints: [JobConstraint]) -> T? { ... }
```

**Comment by arthurdapaz** [comment]
Ok @lucas34 , I have fixed the issue now. :) After deep investigation, the problem is that the generic functions **(getConstraint)** aren't able to retrieve the **persisterConstraint** from **job.info**. On debug I noticed that inside "job" we can recover the persisterConstraint, but inside job.info the persisterConstraint is not added at all

<img width="947" alt="Captura de Tela 2020-12-20 às 01 11 58" src="https://user-images.githubusercontent.com/2267843/102704951-b96aa600-4260-11eb-918f-b4485ab2028e.png">

**Comment by arthurdapaz** [comment]
Created a merge request here: https://github.com/lucas34/SwiftQueue/pull/369

**Comment by lucas34** [comment]
@arthurdapaz  Awesome thanks, I have merged your MR, I'm preparing a 5.0.2 release

**Comment by arthurdapaz** [comment]
> @arthurdapaz Awesome thanks, I have merged your MR, I'm preparing a 5.0.2 release

This issue can be closed now. Since 5.0.2 its already released! Thank you, sir!

**Comment by Narayane** [comment]
Hi,

I confirm that solve my bug, thanks @arthurdapaz for fix.
